### PR TITLE
simplify ./gradlew run-ccs

### PR DESCRIPTION
**Pull request by jakelandis**
This commit reverts the changes introduced via https://github.com/elastic/elasticsearch/pull/89442 / 20ed7e3fd9b492949c86b68ec637028c9f491da6
and re-implements the same goals but with fewer changes. Also included in this commit 
is the ability to change the proxyMode via system property and mention of this
new run task in the testing documentation. 

===

While working https://github.com/elastic/elasticsearch/pull/89563 I found a better pattern that does not require updates to the run task. I think this approach is cleaner especially for debugging.

### CODE REVIEW

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersAware.java [hunk 1]
```diff
@@ -37,8 +37,8 @@ default void useCluster(Provider<ElasticsearchCluster> cluster) {
 
     default void beforeStart() {}
 
-    default void enableDebug(int portOffset) {
-        int debugPort = 5007 + portOffset;
+    default void enableDebug() {
```
Comment (jakelandis) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersAware.java:40]: part of the revert

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/StandaloneRestIntegTestTask.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public WorkResult delete(Object... objects) {
     @Override
     public void beforeStart() {
         if (debugServer) {
-            enableDebug(0);
+            enableDebug();
```
Comment (jakelandis) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/StandaloneRestIntegTestTask.java:107]: part of the revert

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 3]
```diff
@@ -35,16 +35,12 @@ public class RunTask extends DefaultTestClustersTask {
 
     private Boolean debug = false;
```
Comment (jakelandis) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java:37]: all changes are part of the revert

File: build-tools-internal/src/main/groovy/elasticsearch.run-ccs.gradle [hunk 4]
```diff
@@ -23,35 +22,25 @@ def queryingCluster = testClusters.register('querying-cluster') {
...
   doFirst {
+    queryingCluster.get().getNodes().each { node ->
+      if (proxyMode) {
+        node.setting('cluster.remote.my_remote_cluster.mode', 'proxy')
+        node.setting('cluster.remote.my_remote_cluster.proxy_address', "\"${fulfillingCluster.get().getAllTransportPortURI().get(0)}\"")
+      } else {
+        node.setting('cluster.remote.my_remote_cluster.seeds', fulfillingCluster.get().getAllTransportPortURI().collect { "\"$it\"" }.toString())
+      }
+    }
+    queryingCluster.get().restart() //updates the on-disk config
```
Comment (jakelandis) [inline build-tools-internal/src/main/groovy/elasticsearch.run-ccs.gradle:42]: this is the main change. specifically calling .restart() before either actually start updates the configuration on disk

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by jakelandis** [comment]
thanks Mark!

**Pull request merged**